src/src/PluginStructs/P128_data_struct.cpp

```diff
@@ -653,84 +644,94 @@ bool P128_data_struct::plugin_write(struct EventStruct *event,
     NeoPixelSendStatus(event);
   }                                                                            // command neopixel
 
-  if (speed == 0) { mode = On;                                                 // speed = 0 = stop mode
+  if (speed == 0) { mode = P128_modetype::On;                                  // speed = 0 = stop mode
   }
   speed    = (speed > SPEED_MAX || speed < -SPEED_MAX) ? defaultspeed : speed; // avoid invalid values
```

Comment:
Now you need to always make the assignment.
```c++
if (speed > SPEED_MAX || speed < -SPEED_MAX) speed = defaultspeed;
```

Not sure though what the compiler will make of it. Could be the compiled code is exactly the same ;)